docs/guides/app-payments/build-on-mobile.mdx

```diff
@@ -21,46 +19,199 @@ This page details how MobilePay app payments should be implemented in your Andro
 }>
 <TabItem value="android">
 
+## Prerequisites and Assumptions
+
 This guide assumes:
 
 - You already have backend services integrated with [MobilePay APIs](/docs/guides/app-payments/how-it-works).
 - Your app uses [minSdkVersion](https://developer.android.com/training/basics/supporting-devices/platforms.html#sdk-versions) API 23 (Marshmallow, 6.0) or later.
 - You are generally familiar with developing applications on Android.
 
+:::caution Make sure you are compliant with mobile application distribution service rules!
+Using app payments for digital sales might not be allowed by some mobile application distribution services (Google Play, App Store, etc.) and might result in your application being removed. Please review the terms of service for the respective mobile application distribution services you use to ensure that you are compliant with their guidelines.
+:::
+
+## Create Payment
+
+First, you'll need to create payment. To do this you should call your backend service that is implemented as described [here](/docs/payments-refunds/create-payments). This service call should return a response that contains a deep link needed to launch the MobilePay app.
+
+## Starting MobilePay
+
+MobilePay supports navigation via native Activity Result APIs and deep linking.
+
+<Tabs
+  groupId="android-navigation-method"
+  defaultValue="result"
+  values={[
+    { label: 'Result APIs', value: 'result', },
+    { label: 'Deep Links', value: 'deep_link', },
+  ]
+}>
+<TabItem value="result">
+
+Activity Result or `startActivityForResult` APIs use standard android functions and will launch MobilePay in your app process. This is the preferred way when integrating into native android app.
+
+Let's open MobilePay with the deep link received from your backend.
+
+<Tabs
+  groupId="android-language"
+  defaultValue="kotlin"
+  values={[
+    { label: 'Kotlin using Activity Result APIs', value: 'kotlin', },
+    { label: 'Kotlin using startActivityForResult', value: 'activity_result_kotlin', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+
+<TabItem value="kotlin">
+
+```kotlin title="Kotlin using Activity Result APIs (Requires AndroidX Activity 1.2.0 or Fragment 1.3.0) "
+private val mobilePayContract = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+  when (it.resultCode) {
+    SUCCESS_RESULT_CODE -> {}
+    REJECT_RESULT_CODE -> {}
+    FAILED_RESULT_CODE -> {}
+    UNKNOWN_RESULT_CODE -> {}
+  }
+}
+
+// Use the link received from backend to launch MobilePay.
+fun openMobilePay(deepLink: String) {
+  try {
+      val intent = Intent(Intent.ACTION_VIEW).apply {
+          data = Uri.parse(deepLink)
+      }
+      mobilePayContract.launch(intent)
+  } catch (exception: ActivityNotFoundException) {
+      // Inform the user that MobilePay is not installed.
```

Comment:
Are we "allowed" to provide the MP package name here in this documentation? If yes, then we could do:

```suggestion
      // Inform the user that MobilePay is not installed (or even better, direct the user to the package <package> directly in Google Play).
```

Reply:
added links to play stores.